src/production/views.py

```diff
@@ -951,6 +951,87 @@ def edit_galley(request, galley_id, typeset_id=None, article_id=None):
     return render(request, template, context)
 
 
+@typesetter_or_editor_required
+def upload_image_zip(request, galley_id, typeset_id=None, article_id=None):
+    return_url = request.GET.get('return', None)
+
+    if typeset_id:
+        typeset_task = get_object_or_404(
+            models.TypesetTask,
+            pk=typeset_id,
+            assignment__article__journal=request.journal,
+            accepted__isnull=False,
+            completed__isnull=True,
+        )
+        article = typeset_task.assignment.article
+    else:
+        typeset_task = None
+        article = get_object_or_404(
+            submission_models.Article.allarticles,
+            pk=article_id,
+            journal=request.journal
+        )
+
+    galley = get_object_or_404(
+        core_models.Galley,
+        pk=galley_id,
+        article=article,
+    )
+
+    if request.POST and 'zip_file' in request.POST:
+        file = request.FILES.get('file')
+        try:
+            errors = logic.process_zip_file(file, galley, request)
```

Comment:
This is a bit strange to see in python (not so much in go :P )

maybe have `process_zip_file` raise an exception at the end of processing the file and then:
```
Try:
    logic.process_zip_file(file, galley, request)
except BadZipFile:
   [...]
except FoundAdditionalFiles: # or some other name
   [...]
```